pytorch_lightning/callbacks/pruning.py

```diff
@@ -80,77 +79,84 @@ def __init__(
+
+            amount: Quantity of parameters to prune:
 
-            amount: quantity of parameters to prune:
+                - ``float``. Between 0.0 and 1.0. Represents the fraction of parameters to prune.
+                - ``int``. Represents the absolute number of parameters to prune.
+                - ``Callable``. For dynamic values. Will be called every epoch. Should return a value.
 
-                - float, should be between 0.0 and 1.0 and represent the fraction of parameters to prune.
-                - int, it represents the absolute number of parameters to prune.
-                - Callable, the function will be called on every epoch.
+            apply_pruning: Whether to apply pruning.
 
-            make_pruning_permanent: if True then all
-                reparametrization pre-hooks and tensors with mask
-                will be removed on fit end.
+                - ``bool``. Always apply it or not.
+                - ``Callable[[epoch], bool]``. For dynamic values. Will be called every epoch.
 
-            use_lottery_ticket_hypothesis: Wether to use algorithm describes in
-                "The lottery ticket hypothesis" (https://arxiv.org/pdf/1803.03635.pdf)
+            make_pruning_permanent: Whether to remove all reparametrization pre-hooks and apply masks
+                when training ends or the model is saved.
 
-            pruning_dim: if you are using structured pruning method you need
-                to specify dimension.
+            use_lottery_ticket_hypothesis: See "The lottery ticket hypothesis" (https://arxiv.org/pdf/1803.03635.pdf):
 
-            pruning_norm: if you are using ln_structured you need to specify norm.
+                - ``bool``. Whether to apply it or not.
+                - ``Callable[[epoch], bool]``. For dynamic values. Will be called every epoch.
+
+            resample_parameters: Used with ``use_lottery_ticket_hypothesis``. If True, the model parameters will
+                be resampled, otherwise, the exact original parameters will be used.
+
+            pruning_dim: If you are using a structured pruning method you need to specify the dimension.
+
+            pruning_norm: If you are using ``ln_structured`` you need to specify the norm.
+
+            verbose: Whether to log pruning percentage changes.
 
         """
 
         self._use_global_unstructured = use_global_unstructured
         self._parameters_to_prune = parameters_to_prune
         self._use_lottery_ticket_hypothesis = use_lottery_ticket_hypothesis
+        self._resample_parameters = resample_parameters
         self._parameter_names = parameter_names or self.PARAMETER_NAMES
         self._global_kwargs = {}
-        self._initial_parameters_to_prune = None
+        self._original_layers = None
 
-        for param_name in self._parameter_names:
-            if param_name not in self.PARAMETER_NAMES:
+        for name in self._parameter_names:
```

Comment:
IMO, param_name was more explicit.

Reply:
Notice this change is internal, users still pass `param_names`. Using `name` internally follow PyTorch's naming